src/pal/src/misc/corefx.cpp

```diff
@@ -346,3 +348,74 @@ ForkAndExecProcess(
     return success ? 0 : -1;
 }
 
+void CopyStatToFileInfo(struct fileinfo* dst, const struct stat* src)
+{
+    dst->flags = FILEINFO_FLAGS_NONE;
+    dst->mode = src->st_mode;
+    dst->uid = src->st_uid;
+    dst->gid = src->st_gid;
+    dst->size = src->st_size;
+    dst->atime = src->st_atime;
+    dst->mtime = src->st_mtime;
+    dst->ctime = src->st_ctime;
+
+    #if __APPLE__
```

Comment:
I'm not sure, but HAVE_STRUCT_STAT_ST_BIRTHTIME might be another/better option to check.